src/CommandingOfficers/Meridian.java

```diff
@@ -0,0 +1,375 @@
+    {
+      return null;
+    }
+  }
+
+  /** Effectively a wait, but the unit ends up as a different unit at the end of it. */
+  public static class TransformAction implements GameAction
+  {
+    private Transform type;
+    private Path movePath;
+    private XYCoord waitLoc = null;
+    private Unit actor = null;
+
+    public TransformAction(Unit unit, Path path, Transform pType)
+    {
+      actor = unit;
+      movePath = path;
+      if( (null != path) && (path.getPathLength() > 0) )
+      {
+        // Store the destination for later.
+        waitLoc = new XYCoord(path.getEnd().x, path.getEnd().y);
+      }
+      type = pType;
+    }
+
+    @Override
+    public GameEventQueue getEvents(MapMaster gameMap)
+    {
+      // WAIT actions consist of
+      //   MOVE
+      GameEventQueue transformEvents = new GameEventQueue();
+
+      // Validate input.
+      boolean isValid = true;
+      isValid &= null != actor && !actor.isTurnOver;
+      isValid &= (null != movePath) && (movePath.getPathLength() > 0);
+      isValid &= (null != gameMap);
+      if( isValid )
+      {
+        XYCoord movePathEnd = new XYCoord(movePath.getEnd().x, movePath.getEnd().y);
+        Location moveLocation = gameMap.getLocation(movePathEnd);
+        isValid &= (null == moveLocation.getResident()) || (actor == moveLocation.getResident());
+      }
+
+      // Generate events.
+      if( isValid )
+      {
+        if (Utils.enqueueMoveEvent(gameMap, actor, movePath, transformEvents))
+        {
+          transformEvents.add(new UnitTransformEvent(actor, type.destinationType, type.co));
+        }
+      }
+      return transformEvents;
+    }
+
+    @Override
+    public XYCoord getMoveLocation()
+    {
+      return waitLoc;
+    }
+
+    @Override
+    public XYCoord getTargetLocation()
+    {
+      return waitLoc;
+    }
+
+    @Override
+    public ActionType getType()
+    {
+      return ActionType.UNIT;
+    }
+
+    @Override
+    public String toString()
+    {
+      return String.format("[Move %s to %s and transform to %s]", actor.toStringWithLocation(), waitLoc, type.destinationType);
+    }
+
+    @Override
+    public UnitActionType getUnitActionType()
+    {
+      return type;
+    }
+  }
+
+  /** Effectively a wait, but the unit ends up as a different unit at the end of it. */
+  public static class UnitTransformEvent implements GameEvent
+  {
+    private Unit unit;
+    private UnitModel type;
+    private Meridian user;
+
+    public UnitTransformEvent(Unit unit, UnitModel destination, Meridian listener)
+    {
+      this.unit = unit;
+      type = destination;
+      user = listener;
+    }
+
+    @Override
+    public GameAnimation getEventAnimation(MapView mapView)
+    {
+      return null;
+    }
+
+    @Override
+    public void sendToListener(GameEventListener listener)
+    {
+      // listener.receiveUnitDieEvent( this );
+    }
+
+    @Override
+    public void performEvent(MapMaster gameMap)
```

Comment:
Huh. I probably would have just used a one-way `UnitRemodelModifier`.

Reply:
Which, to be fair, doesn't attempt to adjust ammo, but it could.

Reply:
That mildly feels like using a vise when pliers would be sufficient, but that would be a way to re-use code.

Reply:
How so? You are remodeling the unit, after all. And we use `COModifiers` without intent to revert in lots of places.

Reply:
Sure, but this is an event for 1 unit. `COModifiers` are for Commanders, not so much individual units.